### PR TITLE
BFD-2684: Increase minimum AWS file download part size to 200MB

### PR DESCRIPTION
**Pull request by lsmitchell**
**JIRA Ticket:**
[BFD-2684](https://jira.cms.gov/browse/BFD-2684)

**User Story or Bug Summary:**

Attempts to fix a bug where AWS downloads get stuck sometimes.

---

### What Does This PR Do?

Increases the async S3 client configuration for maximum file size before splitting the file into smaller parts to 200MB. This will also increase the chunk size of each download piece to 200MB, which should reduce the number of threads needed for a large download.

Also deletes a KMS key addition to the copy request that was causing issues in prod-sbx; basically if the key is found and added to the request the copy fails, because AWS was expecting some specific header that wasnt added correctly. The kms key addition was unneeded, because the copy request if absent the specific kms key will default to the bucket-level encryption, which is what we want anyway. We'll let AWS handle that automatically instead of explicitly adding it.

Also increases the size of the test/prod-sbx pipeline boxes now that we have an on-demand solution for the ccw pipeline creation.

### What Should Reviewers Watch For?

If you're reviewing this PR, please check for these things in particular:

* Verify all PR security questions and checklists have been completed and addressed.

### What Security Implications Does This PR Have?

Submitters should complete the following questionnaire:

* If the answer to any of the questions below is **Yes**, then **you must** supply a link to the associated Security Impact Assessment (SIA), security checklist, or other similar document in Confluence here: **N/A**

    * Does this PR add any new software dependencies? 
      * [ ] Yes
      * [x] No
    * Does this PR modify or invalidate any of our security controls?
      * [ ] Yes
      * [x] No
    * Does this PR store or transmit data that was not stored or transmitted before?
      * [ ] Yes
      * [x] No

* If the answer to any of the questions below is **Yes**, then please add @StewGoin as a reviewer, and note that this PR **should not be merged** unless/until he also approves it.
    * Do you think this PR requires additional review of its security implications for other reasons?
      * [ ] Yes
      * [x] No

### What Needs to Be Merged and Deployed Before this PR?

This PR cannot be either merged or deployed until the following prerequisite changes have been fully deployed:

* N/A

### Submitter Checklist

I have gone through and verified that...:

* [x] I have named this PR and branch so they are [automatically linked](https://confluence.atlassian.com/adminjiracloud/integrating-with-development-tools-776636216.html) to the (most) relevant Jira issue. Ie: `BFD-123: Adds foo`
* [x] This PR is reasonably limited in scope, to help ensure that:
    1. It doesn't unnecessarily tie a bunch of disparate features, fixes, refactorings, etc. together.
    2. There isn't too much of a burden on reviewers.
    3. Any problems it causes have a small "blast radius".
    4. It'll be easier to rollback if that becomes necessary.
* [x] This PR includes any required documentation changes, including `README` updates and changelog / release notes entries.
* [x] The data dictionary has been updated with any field mapping changes, if any were made.
* [x] All new and modified code is appropriately commented, such that the what and why of its design would be reasonably clear to engineers, preferably ones unfamiliar with the project.
* [x] All tech debt and/or shortcomings introduced by this PR are detailed in `TODO` and/or `FIXME` comments, which include a JIRA ticket ID for any items that require urgent attention.
* [x] Reviews are requested from both:
    * At least two other engineers on this project, at least one of whom is a senior engineer or owns the relevant component(s) here.
    * Any relevant engineers on other projects (e.g. DC GEO, BB2, etc.).
* [x] Any deviations from the other policies in the [DASG Engineering Standards](https://github.com/CMSgov/cms-oeda-dasg/blob/master/policies/engineering_standards.md) are specifically called out in this PR, above.
    * Please review the standards every few months to ensure you're familiar with them.

### CODE REVIEW

**Review by mackec1 — APPROVED** [review]

**Comment by lsmitchell** [comment]
As a side note locally everything works fine even with the old code; this only seemed to fail on prod-sbx so far for a synthetic load. Suspect this may fix it, but can't be sure without a real test.

**Review by billie-rose — DISMISSED** [review]

File: apps/bfd-pipeline/bfd-pipeline-shared-utils/src/main/java/gov/cms/bfd/pipeline/sharedutils/s3/SharedS3Utilities.java [hunk 1]
```diff
@@ -97,7 +96,9 @@ public static S3AsyncClient createS3AsyncClient(Region awsS3Region) {
     if (minioConfig.useMinio) {
       return createS3AsyncMinioClient(awsS3Region, minioConfig);
     }
-    return S3CrtAsyncClient.builder().region(awsS3Region).build();
+    // Split the file download (automatically using the async client) if its >200 MB
+    long mb = 1048576L;
+    return S3AsyncClient.crtBuilder().minimumPartSizeInBytes(200 * mb).region(awsS3Region).build();
```
Comment (mjburling) [inline apps/bfd-pipeline/bfd-pipeline-shared-utils/src/main/java/gov/cms/bfd/pipeline/sharedutils/s3/SharedS3Utilities.java:101]: This seems reasonable. What's the default part size?

I don't think there's any harm in doing this, but it doesn't seem like we're _too_ confident in this change being the solution. I'm building the images based off this PR now and I can deploy the result to `prod-sbx` so we can evaluate it **before** acceptance. What do you think?

Comment (lsmitchell) [reply apps/bfd-pipeline/bfd-pipeline-shared-utils/src/main/java/gov/cms/bfd/pipeline/sharedutils/s3/SharedS3Utilities.java:101]: Yup that seems like a good idea. The default previously was 8MB, so a 400MB file would be split into 50 parts. I'm still not entirely sure what the issue is given that things work fine locally, so a test on the box would be prudent.

**Comment by lsmitchell** [comment]
Testing worked with the changes here in prod-sbx; was able to load all data and it was successfully moved to done.

**Review by mjburling — APPROVED** [review]

**Review by malessi — APPROVED** [review]

**Pull request merged**